### ISSUE TITLE
expression-not-assigned should not be emitted for ternary operator

### ISSUE THREAD

**Issue by sushobhit27** [PyCQA/pylint#2380]
### Steps to reproduce
1. cat sample.py
```python

def goo(var):
    pass

def bar(var):
    pass

def foo():
    goo(True) if True else bar(False)
```
2. $ pylint sample.py --disable=blacklisted-name,unused-argument,using-constant-test

### Current behavior
************* Module sample
sample.py:8:4: W0106: Expression "goo(True) if True else bar(False)" is assigned to nothing (expression-not-assigned)

------------------------------------------------------------------
Your code has been rated at 8.33/10 (previous run: 8.33/10, +0.00)

### Expected behavior
There should be no warning when some functionality has to be executed using ternary operator.
Actuall use case, where this message is being raised: https://github.com/sushobhit27/bimap/blob/8b28e6501b96872c109e0a51e4f7e6081bec0c8f/bimap/bimap.py#L57

### pylint --version output
$ pylint --version
pylint 2.0.0
astroid 2.0.1
Python 3.6.4 (default, Jan  7 2018, 15:53:53)
[GCC 6.4.0]

**Comment by PCManticore** [comment]
@sushobhit27 I think it's reasonable that this check gets emitted here. This would be better written as an if statement, which is a bit clearer and easier to follow.

```
if True:
   run_this()
else:
   run_that()
```

**Comment by PCManticore** [comment]
But in your case you are creating a transient list that just gets thrown away after you're done with the iteration. It's more efficient to just use the `for` loop here, and it goes without saying that it helps with the readability of the code as well.
